inginious/frontend/plugins/code_preview/static/js/code_preview_load.js

```diff
@@ -1,25 +1,131 @@
-function load_code_preview_to_codemirror() {
-    if (location.href.indexOf("/course") > -1 && location.href.split('/').length == 6) {
-        $.get('/api/code_preview/', {
-            task_id: getTaskIdFromUrl(),
-            course_id: getCourseIdFromUrl(),
-            language: getInginiousLanguageForProblemId(getProblemId())
-        }, function write(result) {
-            if (result !== "") {
-                const ks = Object.keys(codeEditors);
-                ks.forEach(element => {
-                    codeEditors[element].setValue(result);
-                });
+function updateCodePreviewFiles() {
+    $.get("/api/grader_generator/test_file_api", {
+        course_id: getCourseId(),
+        task_id: getTaskId(),
+    }, function (files) {
+        let previewFileSelect = $("#code_preview_files");
+
+        previewFileSelect.empty();
+
+        $.each(files, function (index, file) {
+            const fileIsTestCase = file.complete_name.includes(".in") || file.complete_name.includes(".out");
+            if (file.is_directory || fileIsTestCase || file.complete_name === "run") {
+                return;
             }
-        })
+
+            const entry = $("<option>", {
+                "value": file.complete_name,
+                "text": file.complete_name
+            });
+
+            previewFileSelect.append(entry);
+
+        });
+    }, "json");
+}
+
+function addCodePreviewPair() {
+    const language = $("#code_preview_languages > option:selected")[0].value;
+    const file = $("#code_preview_files > option:selected")[0].value;
+
+    if (language === "-1") {
+        return;
+    }
+    if (language in currentCodePreviewPairs) {
+        currentCodePreviewPairs[language] = file;
+        $(`#code_preview_pairs_${language}`).val(file);
+        return;
+    }
+
+    const template = `
+    <div class="row" id="code-preview-${language}-row">
+        <div class="form-group col-xs-12">
+            <div class="col-xs-4">
+                <select class="form-control" readonly>
+                    <option selected>${getLanguageName(language)}</option>
+                </select>
```

Comment:
Maybe change these select element for the file language and set it as a static input just like in the file, since there's only an option to this select.